cmd/restic/integration_test.go

```diff
@@ -1023,6 +1023,46 @@ func TestRebuildIndexAlwaysFull(t *testing.T) {
 	TestRebuildIndex(t)
 }
 
+type appendOnlyBackend struct {
+	restic.Backend
+}
+
+func (b *appendOnlyBackend) Remove(ctx context.Context, h restic.Handle) error {
+	return errors.Errorf("Failed to remove %v", h)
+}
+
+type appendOnlyRepository struct {
+	restic.Repository
+}
```

Comment:
Bikeshedding maybe, but I'm kinda meh on the use of the [preemptive interfaces](https://medium.com/@cep21/preemptive-interface-anti-pattern-in-go-54c18ac0668a) restic.Repository and restic.Cache (see also #2616). I know the former actually gets multiple implementations now, but they still make the code harder to navigate than it could be.

I think the appendOnlyBackend could also be installed inside the existing OpenRepository without the appendOnlyRepository. Just wrap the backend right after it's been constructed. Then the code in global.go can continue to use *repository.Repository and the changes are much more localized.

Reply:
I've replace the repository hook with a backend hook. This removes the code churn which modified everything to use `restic.Repository`.